From 68ea8e24be31cf5e312db6cc35929ec56c3ec65e Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Sun, 17 Nov 2024 12:49:40 +0100
Subject: [PATCH] BACKPORT: Fix build with Python 3.13

Remove includes for headers which are already transitively included by Python.h and fail to be directly included in Python 3.13 without importing some c++ headers first
---
 CMakeLists.txt         | 2 +-
 parser/python_header.h | 5 -----
 2 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9e87281..8d507038 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,7 @@ add_definitions( -DTRANSLATION_DOMAIN=\"kdevpython\" )
 
 set(Python3_USE_STATIC_LIBS FALSE)
 set(Python3_FIND_STRATEGY VERSION)
-find_package(Python3 3.4.3...<3.13 COMPONENTS Interpreter Development REQUIRED)
+find_package(Python3 3.4.3...<3.14 COMPONENTS Interpreter Development REQUIRED)
 
 configure_file(kdevpythonversion.h.cmake "${CMAKE_CURRENT_BINARY_DIR}/kdevpythonversion.h" @ONLY)
 
diff --git a/parser/python_header.h b/parser/python_header.h
index 90dea6a5..6fef957a 100644
--- a/parser/python_header.h
+++ b/parser/python_header.h
@@ -13,11 +13,6 @@
 
 #include <language/duchain/duchainlock.h>
 
-#include "pyport.h"
-#ifndef _WIN32
-#include "pyconfig.h"
-#endif
-
 #include "Python.h"
 
 #include "ast.h"

